### PR TITLE
feat: add `client.getLedgerIndex`

### PR DESCRIPTION
**Pull request by mvadari**
## High Level Overview of Change

This PR adds a new method to the client, `client.getLedgerIndex`, which gets the ledger index of the most recent validated ledger.

### Context of Change

Requested by @mDuo13. Makes it much easier to get a ledger index, which you have to do often.

### Type of Change

- [x] New feature (non-breaking change which adds functionality)

## Test Plan

CI passes. Wrote a test for the new function.

### CODE REVIEW

File: test/client/getLedgerIndex.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import rippled from '../fixtures/rippled'
+import { setupClient, teardownClient } from '../setupClient'
+
+describe('client.getLedgerIndex', function () {
+  beforeEach(setupClient)
+  afterEach(teardownClient)
+
+  it('getLedgerIndex', async function () {
+    this.mockRippled.addResponse('ledger', rippled.ledger.normal)
+    const fee = await this.client.getLedgerIndex()
```
Comment (mDuo13) [inline test/client/getLedgerIndex.ts:12]: Uh, `fee` is probably a misleading variable name here, right?

File: test/integration/setup.ts [hunk 2]
```diff
@@ -16,10 +16,7 @@ export async function suiteClientSetup(this: Mocha.Context): Promise<void> {
   // two times to give time to server to send `ledgerClosed` event
   // so getLedgerVersion will return right value
   await ledgerAccept(this.client)
-  const response = await this.client.request({
-    command: 'ledger',
-    ledger_index: 'validated',
-  })
+  const response = await this.client.getLedgerIndex()
   const ledgerVersion = response.result.ledger_index
   this.startLedgerVersion = ledgerVersion
```
Comment (mDuo13) [inline test/integration/setup.ts]: Looks like this part still needs to be consolidated?

File: test/integration/setup.ts [hunk 3]
```diff
@@ -16,10 +16,7 @@ export async function suiteClientSetup(this: Mocha.Context): Promise<void> {
   // two times to give time to server to send `ledgerClosed` event
   // so getLedgerVersion will return right value
```
Comment (mDuo13) [inline test/integration/setup.ts:17]: Probably should update this comment (`getLedgerVersion` is not the right name at least)

**Review by mDuo13 — CHANGES_REQUESTED** [review]
Some minor nitpicks, but the code works as expected. :tada:

File: test/client/getLedgerIndex.ts (hunk 1 above)
Comment (mvadari) [reply test/client/getLedgerIndex.ts:12]: whoops, copy-paste error

File: test/integration/setup.ts (hunk 2 above)
Comment (mvadari) [reply test/integration/setup.ts]: yep, that was an error, already fixed

File: test/integration/setup.ts [hunk 4]
```diff
@@ -13,9 +13,6 @@ export async function suiteClientSetup(this: Mocha.Context): Promise<void> {
   await setupClient.bind(this)(serverUrl)
   await ledgerAccept(this.client)
   this.newWallet = generateXAddress({ includeClassicAddress: true })
-  // two times to give time to server to send `ledgerClosed` event
-  // so getLedgerVersion will return right value
-  await ledgerAccept(this.client)
```
Comment (mDuo13) [inline test/integration/setup.ts:18]: Not sure what `ledgerAccept` was there for before. Was this related to [ledger_accept](https://xrpl.org/ledger_accept.html) for stand-alone mode, or just a naming coincidence?

**Review by mDuo13 — APPROVED** [review]

File: test/integration/setup.ts (hunk 4 above)
Comment (mvadari) [reply test/integration/setup.ts:18]: It's a helper function that basically runs `ledger_accept` in the standalone container (it's in `test/integration/utils.ts`). I guess it was necessary when `getLedgerVersion` listened to a `ledger` subscription.

**Review by mukulljangid — APPROVED** [review]

**Review by natenichols — APPROVED** [review]

**Pull request merged**